catalog/glue.go

```diff
@@ -0,0 +1,186 @@
+
+		if tblsRes.NextToken == nil {
+			break
+		}
+
+		params.NextToken = tblsRes.NextToken
+	}
+
+	return icebergTables, nil
+}
+
+// LoadTable loads a table from the catalog table details.
+//
+// The identifier should contain the Glue database name, then glue table name.
+func (c *GlueCatalog) LoadTable(ctx context.Context, identifier table.Identifier, props map[string]string) (*table.Table, error) {
+	database, tableName, err := identifierToGlueTable(identifier)
+	if err != nil {
+		return nil, err
+	}
+
+	if props == nil {
+		props = map[string]string{}
+	}
+
+	location, err := c.getTable(ctx, database, tableName)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: consider providing a way to directly access the S3 iofs to enable testing of the catalog.
+	iofs, err := io.LoadFS(props, location)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load table %s.%s: %w", database, tableName, err)
+	}
+
+	icebergTable, err := table.NewFromLocation([]string{tableName}, location, iofs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table from location %s.%s: %w", database, tableName, err)
+	}
+
+	return icebergTable, nil
+}
+
+func (c *GlueCatalog) CatalogType() CatalogType {
+	return Glue
+}
+
+// GetTable loads a table from the Glue Catalog using the given database and table name.
+func (c *GlueCatalog) getTable(ctx context.Context, database, tableName string) (string, error) {
+	tblRes, err := c.glueSvc.GetTable(ctx,
+		&glue.GetTableInput{
+			DatabaseName: aws.String(database),
+			Name:         aws.String(tableName),
+		},
+	)
+	if err != nil {
+		if errors.Is(err, &types.EntityNotFoundException{}) {
+			return "", fmt.Errorf("failed to get table %s.%s: %w", database, tableName, ErrNoSuchTable)
+		}
+		return "", fmt.Errorf("failed to get table %s.%s: %w", database, tableName, err)
+	}
+
+	if tblRes.Table.Parameters["table_type"] != "ICEBERG" {
+		return "", errors.New("table is not an iceberg table")
+	}
+
+	return tblRes.Table.Parameters["metadata_location"], nil
+}
+
+func identifierToGlueTable(identifier table.Identifier) (string, string, error) {
+	if len(identifier) != 2 {
+		return "", "", fmt.Errorf("invalid identifier, missing database name: %v", identifier)
+	}
+
+	return identifier[0], identifier[1], nil
+}
+
+func identifierToGlueDatabase(identifier table.Identifier) (string, error) {
+	if len(identifier) != 1 {
+		return "", fmt.Errorf("invalid identifier, missing database name: %v", identifier)
+	}
+
+	return identifier[0], nil
+}
+
+// GlueTableIdentifier returns a glue table identifier for an iceberg table in the format [database, table].
+func GlueTableIdentifier(database string, tableName string) table.Identifier {
+	return []string{database, tableName}
+}
+
+// GlueDatabaseIdentifier returns a database identifier for a Glue database in the format [database].
+func GlueDatabaseIdentifier(database string) table.Identifier {
+	return []string{database}
+}
```

Comment:
Could these be made generic rather than `Glue` specific? Any catalog specific logic should happen by manipulating the `Identifier` internally to that catalog, right?

Reply:
@zeroshade  I wanted to provide some helpers for handling this identifier tuple for Glue to make it SUPER clear and use the terminology of Glue, rather than providing some vague external table mapping indexes in a tuple to a given catalogues expectation. 

Users of Glue should be able to clearly see their terminology in the API as it will get them running faster, and make troubleshooting easier.

The use of a tuple for identifier feels a bit odd in a Go API, ideally I think these semantics would be in the identifier using a struct with a dash of generics. Catalogs could then extend this when needed but yeah

Reply:
Okay, that makes sense. We can always adjust later if we find we're duplicating things.